### ISSUE TITLE
Jetpack Search: improve tab order for keyboard navigation

### ISSUE THREAD

**Issue by bluefuton** [Automattic/jetpack#19035]
#### Steps to reproduce the issue

1. Go to a site with Instant Search enabled and open the search modal.
2. Use the keyboard to navigate through the page with Tab.

#### What I expected
The tab order reflects the presented modal, and I'm able to get to search results and filters easily.

#### What happened instead
The tab order when you have a search with results currently is:

input > clear > search > sort relevance > sort newest > sort oldest > close modal > results > filters

If 'clear filter' is present, it doesn't seem possible to tab past that.

If I'm selecting filters, it's not clear how to clear the filters easily.

If I'm at the end of the results, it's not clear how to close the modal.

There's no focus style (outline) on the close button.

**Comment by a8ck3n** [comment]
Note that in Firefox, the Search options and results aren't recognized as part of the tab-loop. This should be partially addressed once #25163 is merged.

https://github.com/Automattic/jetpack/pull/25163

**Comment by annacmc** [comment]
I've unassigned this for the moment as after further investigation, I think it would help to look at this after some of the other tab-order related issues are resolved. I suspect a lot of this will be resolved then.

**Comment by a8ck3n** [comment]
I think most of the issues outlined here have been addressed now.

> The tab order when you have a search with results currently is:
> input > clear > search > sort relevance > sort newest > sort oldest > close modal > results > filters

The new tab order is:

input → clear → close → sort options → sort results → filters → secondary close button → repeat

> If 'clear filter' is present, it doesn't seem possible to tab past that.

Addressed in #25342.

> If I'm selecting filters, it's not clear how to clear the filters easily.

Hitting the Space or Enter key will clear the selection or you can tab back to the "Clear filters" button now that tabbing has been addressed. Might be nice to have a shortcut key assigned too but we do not currently have that.

> If I'm at the end of the results, it's not clear how to close the modal.

Handled as part of #25288. We have a secondary close button (hidden until active) that acts as the anchor for the tab loop. If you tab past this button, you go back to the input field.

> There's no focus style (outline) on the close button.

Addressed by @annacmc in #25260.

Would probably be nice to get another round of a11y testing in now that all our changes have landed but the initial work has improved the keyboard navigation substantially.
